### DISCUSSION TITLE
Process for setting up a new project using this as a template

### DISCUSSION THREAD

**Discussion by pauldaustin** [bcgov/quickstart-openshift#1159] [General]
@DerekRoberts,

Continuing our discussion here. I have added a pull request for the change.

**Comment by DerekRoberts** [comment]
@pauldaustin Thanks for the contribution!  :)

It's SUPPOSED to verify that the fallback_tag is usable, so people can start new projects without the need for one to exist.  Looks like that's changed, which is why you were unfortunately blocked.  :(

I've created an issue to get this fixed, preferably in a hurry.

Issue: https://github.com/bcgov/quickstart-openshift/issues/1158

PR: https://github.com/bcgov-nr/action-builder-ghcr/pull/19

**Reply by pauldaustin** [reply]
@DerekRoberts see revised PR that splits out the running of the command from the checking to see if it was successful.

Note that GitHub returns 403 not 401 if there are no packages for a repository.

**Reply by DerekRoberts** [reply]
@pauldaustin That's great!  Thanks a ton for the catch and quick fix.  Let's get that merged and a hotfix release out right away.  :)

**Reply by DerekRoberts** [reply]
@pauldaustin You're the star of our latest release!  :D
https://github.com/bcgov-nr/action-builder-ghcr/releases/tag/v1.1.2

**Reply by pauldaustin** [reply]
@DerekRoberts I'm a big fan of avoiding using a UI to do things when there are IAC (Infrastructure as Code) or command line ways to do things.

Here is a set of bash commands that would create a new repo from the template, pull the OC_TOKEN from openshift and setup and environment variables and secrets. I can create a PR for the documentation. 

There might also be a way to setup all the settings as well.

Pre-requisites:
* GitHub CLI
* Openshift CLI
* Member of the bcgov organization

```bash
repo=bcgov/[repository name]
ocServer=https://api.silver.devops.gov.bc.ca:6443
ocNamespace=[Openshift Namespace]

oc login --token=[OC token] --server=https://api.silver.devops.gov.bc.ca:6443
gh repo create $repo --public --template=bcgov/quickstart-openshift
servicePrincipal=`(oc get secrets -n $ocNamespace -o name 2> /dev/null) |grep pipeline-token | sed 's/secret\///'`
(oc get secret $servicePrincipal -n $ocNamespace -o go-template --template="{{.data.token|base64decode}}" 2> /dev/null) | \
  gh secret set OC_TOKEN --repo $repo 
gh variable set OC_SERVER --repo $repo --body $ocServer
gh variable set OC_NAMESPACE --repo $repo --body $ocNamespace
```

**Reply by DerekRoberts** [reply]
@pauldaustin Damn, that's great!  Please do create a PR.  If not, I'd be hard pressed not copy and paste this right in.  ;)

**Reply by DerekRoberts** [reply]
You now have write access to create branches.  PRs created from those run with higher permissions than forks.  No pressure, but those contributions would be appreciated!

New issue for CLI steps.  https://github.com/bcgov/quickstart-openshift/issues/1166
